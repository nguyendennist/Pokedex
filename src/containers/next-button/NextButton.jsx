import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Button from 'components/button/Button.jsx';
import incrementOffset from 'actions/pokemon/increment-offset/increment-offset';

const NextButton = props => {
	const nextPage = () => {
		if (props.pokemons.length === 20) {
			props.handleOffsetChange();
			setTimeout(() => props.fetchPokemons(), 250);
		}
	};

	return (<Button className='next-button' name='next' onClick={nextPage} />);
};

NextButton.propTypes = {
	fetchPokemons: PropTypes.func.isRequired,
	handleOffsetChange: PropTypes.func.isRequired,
	pokemons: PropTypes.array.isRequired
};

const mapDispatchToProps = dispatch => ({
	handleOffsetChange: () => {
		dispatch(incrementOffset(20));
	}
});

const mapStateToProps = state => ({
	pokemons: state.pokemons
});

export default connect(mapStateToProps, mapDispatchToProps)(NextButton);
