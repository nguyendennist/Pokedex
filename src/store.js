import { createStore } from 'redux';
import pokemon from 'reducers/pokemon/pokemon';

const store = createStore(pokemon);

export default store;
