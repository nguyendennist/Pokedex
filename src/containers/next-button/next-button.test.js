import React from 'react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import store from 'store';
import NextButton from './NextButton.jsx';

describe('unit tests', () => {
  test('should be defined', () => {
    expect(NextButton).toBeDefined();
  });
});

describe('ui tests', () => {
  test('should render correctly', () => {
    const tree = create(
      <Provider store={store}>
        <NextButton getPokemons={() => {}} />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
