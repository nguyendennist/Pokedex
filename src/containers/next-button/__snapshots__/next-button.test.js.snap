// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui tests should render correctly 1`] = `
<div
  className="next-button"
  onClick={[Function]}
>
  <p>
    next
  </p>
</div>
`;
